src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -231,14 +231,26 @@ private boolean isUnusedImport(String imprt) {
     private void processIdent(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         final int parentType = parent.getType();
-        if (parentType != TokenTypes.DOT
+        if ((parentType != TokenTypes.DOT
             && parentType != TokenTypes.METHOD_DEF
             || parentType == TokenTypes.DOT
-                && ast.getNextSibling() != null) {
+                && ast.getNextSibling() != null)
+                && !innerInterfaceOrClass(parentType)) {
```

Comment:
`innerInterfaceOrClass` checks parentType, but in line 236 similar check is performed. They can be merged in one method

Reply:
I didn't get what are you trying to say. Shall I do the conditional checking here instead of creating another method?

Reply:
Sorry, I missed order of checks, it is ok